var/spack/repos/builtin/packages/survey/package.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Survey(CMakePackage):
+    """Survey is a high-level performance tool product from Trenza, Inc.
+       The survey collector/analytics framework is a new generation,
+       high-level, light-weight multi-platform Linux tool set that
+       targets metric collection for high-level performance analysis
+       of applications running on both single node and on large-scale
+       platforms, including the Cray platforms.
+
+       The collector is designed to work on sequential, MPI, OpenMP,
+       and hybrid codes and directly leverages several interfaces
+       available for tools inside current MPI implementations including:
+       MPICH, MVAPICH, MPT, and OpenMPI. It also supports multiple
+       architectures and has been tested on machines based on Intel,
+       AMD, ARM, and IBM P8/9 processors and integrated GPUs.
+
+       Survey is a licensed product – source not openly available.
+       To access the survey source and build with spack please contact:
+       Trenza Inc. via: dmont@trenzasynergy.com or
+                        jeg@trenzasynergy.com
+    """
+
+    homepage = "http://www.trenzasynergy.com"
+    git      = "git@gitlab.com:trenza/survey.git"
+
+    maintainers = ['jgalarowicz']
+
+    version('master', branch='master')
+    version('1.0.1', branch='1.0.1')
+    version('1.0.0', branch='1.0.0')
+
+    variant('mpi', default=False,
+            description="Enable mpi, build MPI data collector")
+
+    # must have cmake at 3.12 or greater to find python3
+    depends_on('cmake@3.12:', type='build')
+
+    # for collectors
+    depends_on("libmonitor@2021.04.27+commrank", type=('build', 'link', 'run'))
+
+    depends_on("papi@5:", type=('build', 'link', 'run'))
+    depends_on("gotcha@master", type=('build', 'link', 'run'))
+    depends_on("llvm-openmp@9.0.0", type=('build', 'link', 'run'))
+
+    # MPI Installation
+    depends_on("mpi", when="+mpi")
+
+    depends_on("python@3:", type=('build', 'link', 'run'))
+    depends_on("py-setuptools", type='run')
```

Comment:
setuptools is usually only needed at build-time, sometimes needed at build/run-time, but never only at run-time. Check your installation prefix for imports of `setuptools` or `pkg_resources`. If you don't see any, this should be:
```suggestion
    depends_on("py-setuptools", type='build')
```

Reply:
Here is setup.py:
```from setuptools import setup
import subprocess

setup(
    name='survey',
    package_dir={ '': '/home/jgalarowicz/survey' },
    packages=['pysurvey'],
    package_data={'pysurvey':['data/papi.txt', 'data/formulas.txt', 'data/cpu_table.json', 'data/top_down.json', 'data/config.yaml']},
    scripts=['/home/jgalarowicz/survey/survey', '/home/jgalarowicz/survey/survey_viewer'],
    py_modules=['plugin_collection', 'plugins.ior', 'plugins.nvidia'],
    license='proprietary',
    description='Python scripts for the survey project',
    install_requires=['pandas', 'psutil', 'sqlalchemy', 'pyyaml']
)

# Retrieve the survey version from the version.py file.
out = subprocess.Popen(['python', 'pysurvey/version.py'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
stdout,stderr = out.communicate()
version = str(stdout)
```
That appears to be the only file from your list that we are using:
```jgalarowicz@viento:~/survey$ find . -name pyproject.toml
jgalarowicz@viento:~/survey$ find . -name setup.py
./build/setup.py
jgalarowicz@viento:~/survey$ cat ./build/setup.py
from setuptools import setup
import subprocess

setup(
    name='survey',
    package_dir={ '': '/home/jgalarowicz/survey' },
    packages=['pysurvey'],
    package_data={'pysurvey':['data/papi.txt', 'data/formulas.txt', 'data/cpu_table.json', 'data/top_down.json', 'data/config.yaml']},
    scripts=['/home/jgalarowicz/survey/survey', '/home/jgalarowicz/survey/survey_viewer'],
    py_modules=['plugin_collection', 'plugins.ior', 'plugins.nvidia'],
    license='proprietary',
    description='Python scripts for the survey project',
    install_requires=['pandas', 'psutil', 'sqlalchemy', 'pyyaml']
)

# Retrieve the survey version from the version.py file.
out = subprocess.Popen(['python', 'pysurvey/version.py'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
stdout,stderr = out.communicate()
version = str(stdout)
jgalarowicz@viento:~/survey$ find . -name setup.cfg
jgalarowicz@viento:~/survey$ find . -name equirements.txt
jgalarowicz@viento:~/survey$ 
```

Reply:
In that case I don't think python-dateutil or pbr are needed, right? Unless they are missing from `setup_requires`/`install_requires`.

Reply:
@adamjstewart I'm checking with the team on the above (pbr and dateutils).  Thanks for your help!